### PR TITLE
Fix error when closing Attach Node Script window

### LINKED ISSUE #34471: Attempt to disconnect signal 'popup_hide' while in emission callback.
**Godot version:**
3.2.beta.custom_build. 981c6aa10

**OS/device including version:**
Ubuntu 19.10
**Issue description:**
Error
```
Attempt to disconnect signal 'popup_hide' while in emission callback. Use CONNECT_DEFERRED (to be able to safely disconnect) or CONNECT_ONESHOT (for automatic disconnection) as connection flags.
````
**Steps to reproduce:**
Try to connect signal(happens rarely) via Node dock
or
Create new node, and create/attach script to it
jfile

### PR DESCRIPTION
**Pull request by volzhs**
Fix #34471

### CODE REVIEW

File: editor/scene_tree_dock.cpp [hunk 1]
```diff
@@ -2616,7 +2616,7 @@ void SceneTreeDock::attach_script_to_selected(bool p_extend) {
 	}
 
 	script_create_dialog->connect("script_created", this, "_script_created");
-	script_create_dialog->connect("popup_hide", this, "_script_creation_closed");
+	script_create_dialog->connect("popup_hide", this, "_script_creation_closed", varray(), CONNECT_DEFERRED);
```
Comment (akien-mga) [inline editor/scene_tree_dock.cpp]: I guess it could be `CONNECT_ONESHOT`, and then the `disconnect` can be removed?

**Comment by akien-mga** [comment]
I wonder if it's really needed to do these disconnects at all in the first place. `popup_hide` is only used here to disconnect the `script_created` signal, but do we need it disconnected?

The only reason I can think of would be that this `ScriptCreateDialog` instance is also reused in `ScriptEditor`, where it's connected but there never disconnected:
```
editor/plugins/script_editor_plugin.cpp
296:void ScriptEditor::_script_created(Ref<Script> p_script) {
3133:   ClassDB::bind_method("_script_created", &ScriptEditor::_script_created);
3415:   script_create_dialog->connect("script_created", this, "_script_created");
```

If it works seemlessly to re-connect the signal without having disconnected it first, then we could simply get rid of the disconnect callback.

**Comment by akien-mga** [comment]
> The only reason I can think of would be that this `ScriptCreateDialog` instance is also reused in `ScriptEditor`, where it's connected but there never disconnected:

Actually each class has a separate instance of `ScriptCreateDialog`, so there's no "reconnection" as I mentioned above. So maybe we can just connect the `script_created` signal in the constructor and then leave it be.

**Comment by akien-mga** [comment]
> Actually each class has a separate instance of `ScriptCreateDialog`, so there's no "reconnection" as I mentioned above. So maybe we can just connect the `script_created` signal in the constructor and then leave it be.

It seems that's what the code used to do, but it was changed in #30196 by @LikeLakers2 for the sake of editor plugins that call `ScriptCreateDialog` themselves. Not sure if that PR was the best approach, but it does make some sense so I guess for now we should just stick to making the `popup_hide` callback a oneshot.

**Comment by LikeLakers2** [comment]
@akien-mga I would recommend having multiple `ScriptCreateDialog` instances if we can code it that way, so long as doing so won't clash with the others. The only reason I haven't done it myself yet is because... I kinda felt it might make a mess of the code when all I wanted to do was fix a bug. Plus it wasn't related to the bugfix. Also...

> Not sure if that PR was the best approach

Honestly, I'm not sure if I went the right approach either. If you want to revert my PR and implement it differently, go right ahead, I won't be offended. (Although I will admit that, looking over my PR again, I do feel a little stupid... but oh well)

File: editor/scene_tree_dock.cpp (hunk 1 above)
Comment (akien-mga) [reply editor/scene_tree_dock.cpp]: Ping.

Comment (akien-mga) [reply editor/scene_tree_dock.cpp]: Force pushed an amend that does that.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**